components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -341,3 +341,197 @@ func Unzip(ctx context.Context, mst storage.MigrationStorage, result pipeline.Re
 	}
 	return result, nil
 }
+
+// ParseOrgUserAssociation  sync the automate org users with chef server org users
+func ParseOrgUserAssociation(ctx context.Context, st storage.Storage, result pipeline.Result) (pipeline.Result, error) {
+	log.Info("Starting with the parsing org user association for migration id :", result.Meta.MigrationID)
+	var orgUserAssociations []pipeline.OrgsUsersAssociations
+	var err error
+	orgUserAssociations, err = getActionForOrgUsers(ctx, st, result)
+	if err != nil {
+		log.Errorf("Unable to parse org user association for migration id : %s : %s", result.Meta.MigrationID, err.Error())
+		return result, err
+	}
+	result.ParsedResult.OrgsUsers = append(result.ParsedResult.OrgsUsers, orgUserAssociations...)
+	log.Info("Completed with the parsing org user association for migration id :", result.Meta.MigrationID)
+	return result, nil
+}
+
+func getActionForOrgUsers(ctx context.Context, st storage.Storage, result pipeline.Result) ([]pipeline.OrgsUsersAssociations, error) {
+	orgUserAssociations := make([]pipeline.OrgsUsersAssociations, 0)
+	var userAssociations []pipeline.UserAssociation
+	orgPath := path.Join(result.Meta.UnzipFolder, "organizations")
+	for _, org := range result.ParsedResult.Orgs {
+		log.Info("Getting actions for org id", org.Name)
+		chefServerOrgUsers, err := getChefServerOrgUsers(org.Name, orgPath)
```

Comment:
We can call the DB only in case of the org is an existing org. Can put it under the else block.

Reply:
@kalroy this function returns the users from back up zip not from the automate DB

Reply:
I meant the one at L374

Reply:
okay. Did the changes, please check